WeatherMonitoringSystem/Domain/DataParsers/WeatherParserStrategy.cs

```diff
@@ -0,0 +1,27 @@
+﻿using WeatherMonitoringSystem.Core;
+using WeatherMonitoringSystem.DataParsers;
+
+public class WeatherParserStrategy
+{
+    private IWeatherDataParser _parser;
+
+    public void SetParser(IWeatherDataParser parser)
+    {
+        _parser = parser ?? throw new ArgumentNullException(nameof(parser));
+    }
+
+    public WeatherData GetWeatherData(string inputData)
+    {
+        if (_parser == null)
+            throw new InvalidOperationException("Parser has not been set.");
+
+        try
+        {
+            return _parser.Parse(inputData);
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"Error parsing JSON/XML: {ex.Message}");
```

Comment:
lets avoid using the generic `Exception` type, since if the caller wanna catch `Exception` it won't know if this is from this method or a different method, we would better throw a specific type of exceptions , if nothing from the existing exception types make  sense, just do a new custom exception for your case (i.e. `XmlJsonParseException`).